arma-rs/src/context/mod.rs

```diff
@@ -4,42 +4,60 @@ use std::sync::Arc;
 
 use crossbeam_queue::SegQueue;
 
-use crate::{IntoArma, State, Value};
+use crate::{IntoArma, Value};
 
 mod arma;
+mod global;
+mod group;
 
 pub use arma::*;
+pub use global::GlobalContext;
+pub use group::GroupContext;
 
 /// Contains information about the current execution context
 pub struct Context {
+    global: GlobalContext,
+    group: Option<GroupContext>,
```

Comment:
I do not like this returning Option. This should instead ideally be represented by the type system, but that would be a bit awkward.

I'd rather it just return a `GroupContext` all the time, and it is just empty when called from a root function.

It will remove a whole bunch of useless `unwrap` calls

Reply:
Addressed by your own PR [fork #2](https://github.com/Eathox/arma-rs/pull/2), thanks for the help!